Bearded.UI/Navigation/DependencyResolver.cs

```diff
@@ -17,7 +20,19 @@ public T Resolve<T>()
             return (T) dict[typeof(T)];
         }
 
-        public bool TryResolve<T>(out T value)
+        public bool TryResolve<T>([NotNullWhen(returnValue: true)] out T? value) where T : class
+        {
+            if (dict.TryGetValue(typeof(T), out var obj))
+            {
+                value = (T) obj;
+                return true;
+            }
+
+            value = null;
```

Comment:
nit: I find `value = default` a bit nicer

Reply:
I think that's a different semantic meaning, and being explicit that we're returning null here is fine and even good.